system/include/emscripten/threading.h

```diff
@@ -7,6 +7,11 @@
 extern "C" {
 #endif
 
+// Returns true if the current browser is able to spawn threads with pthread_create(), and the compiled page was build with
+// threading support enabled. If this returns 0, calls to pthread_create() will fail with return code EAGAIN.
```

Comment:
The documentation for http://man7.org/linux/man-pages/man3/pthread_create.3.html here states "EAGAIN: Insufficient resources to create another thread." and "EAGAIN: A system-imposed limit on the number of threads was encountered.", but it doesn't list ENOSYS at all. I'm thinking "threads not supported" vs "no more threads available" are elegant to handle in the same way. The function `emscripten_has_threading_support()` can be used to detect whether threading is supported at all (without having to create a thread as a side effect).